### PR TITLE
2609 paneu inverse validator

### LINKED ISSUE #2609: panEuropean: Inverse validations logic
The validation message should be: The sum of subcategories is not equal to Total forest ara (1456.46)

Duplicated error messages: Alb forest area Table growing stock total forest 2020:

### PR DESCRIPTION
**Pull request by persas**
Closes #2609

### CODE REVIEW

File: src/client/pages/AssessmentSection/DataTable/Table/DataValidations/DataValidations.tsx [hunk 1]
```diff
@@ -14,6 +14,21 @@ type Props = {
   table: Table
 }
 
+// @ts-ignore
```
Comment (minotogna) [inline src/client/pages/AssessmentSection/DataTable/Table/DataValidations/DataValidations.tsx]: Better not to use any , and avoid ts-ignore

File: src/meta/expressionEvaluator/functions/validatorEqualToSum.ts [hunk 2]
```diff
@@ -10,14 +10,29 @@ export const validatorEqualToSum: ExpressionFunction<Context> = {
   name: 'validatorEqualToSum',
   minArity: 2,
   executor: () => {
-    return (value: string, otherValues: Array<string>): NodeValueValidation => {
+    return (
+      value: string,
+      otherValues: Array<string>,
```
Comment (minotogna) [inline src/meta/expressionEvaluator/functions/validatorEqualToSum.ts:15]: What is otherValues ?

File: src/i18n/resources/en.js [hunk 3]
```diff
@@ -1063,6 +1063,8 @@ The FRA team
     forestSumAreaExceedsExtentOfForest:
       'The sum of the reported FRA categories exceeds the forest area reported in table 1a',
     valueEqualToSum: 'Total value should be equal to sum of subcategories',
+    valueEqualToSumParent:
+      'The sum of subcategories {{subcategories}} is not equal to parent: {{parentVariable}} {{parentCol}} {{parentTable}} <{{valueRounded}}>',
```
Comment (minotogna) [inline src/i18n/resources/en.js]: I would remove the word parent and the : after

File: src/test/sectionSpec/panEuHelpers/updatedDataCol.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        colName: string
+      }
+    }
+    validateFns?: {
+      [key: string]: string[]
+    }
+  }
+}
+
+export const updatedDataCol = (dataCol: DataType[], colName: string, variableName?: string): DataType[] =>
```
Comment (minotogna) [inline src/test/sectionSpec/panEuHelpers/updatedDataCol.ts:20]: What’s happening here ? Why not the getCol anymore ?

File: src/meta/expressionEvaluator/functions/validatorEqualToSum.ts (hunk 2 above)
Comment (persas) [reply src/meta/expressionEvaluator/functions/validatorEqualToSum.ts:15]: otherValues is the array of values to compare (usually, the subcategories). I can rename it to childValues if it is clearer

File: src/test/sectionSpec/panEuHelpers/updatedDataCol.ts (hunk 4 above)
Comment (persas) [reply src/test/sectionSpec/panEuHelpers/updatedDataCol.ts:20]: I figured I was using the same function over and over, and I decided to get it out in a more generic way, and reuse it

File: src/meta/expressionEvaluator/functions/validatorEqualToSum.ts (hunk 2 above)
Comment (minotogna) [reply src/meta/expressionEvaluator/functions/validatorEqualToSum.ts:15]: Aaa ok , let me think a bit about this . No worries for now . 
I'll resolve this comment .

File: src/client/pages/AssessmentSection/DataTable/Table/DataValidations/DataValidations.tsx [hunk 5]
```diff
@@ -14,6 +14,20 @@ type Props = {
   table: Table
 }
 
+const translateIfKey = (t: TFunction, text: any): string => {
```
Comment (minotogna) [inline src/client/pages/AssessmentSection/DataTable/Table/DataValidations/DataValidations.tsx]: Please do not use any . Can it be string | Array<string>?

**Review by minotogna — APPROVED** [review]

**Pull request merged**